### DISCUSSION TITLE
Help with my Fund me.sol smart contract

### DISCUSSION THREAD

**Discussion by Joel-eth** [Cyfrin/foundry-full-course-cu#1604] [General]
I am currently working on a Fundme smart contract and I need to use price feeds to set a minimum amount of deposit in USD. Where can I find the Solidity source code for the Chainlink's AggregatorV3 interface so I can deploy it for my ABI array or is there another way I can go about this?

**Comment by EngrPips** [comment]
Hello @Jostar-dev , are you building this out yourself or you are following Patrick course ?. If you are building it out yourself then i must say i understand you getting stuck and you should follow this link `https://updraft.cyfrin.io/courses/foundry/foundry-fund-me/fund-me-project-setup?lesson_format=video` to Patrick course where he would hand hold you and build a sophisticated `FundMe` smart contract

**Reply by Joel-eth** [reply]
I'm following patrick's youtube course

**Reply by EngrPips** [reply]
If you are following his course he teaches how to do everything step by step so i would say just follow him step by step. and if you have any issue where maybe something works for him in the course and didn't work for you, you should check this link `https://github.com/Cyfrin/foundry-full-course-f23/blob/main/chronological-updates.md` to see what have changed for the section and what you need to do to get yourself unblocked

**Reply by Joel-eth** [reply]
I've gone through it but I haven't seen a response to the change in the location and structure of the source code for the AggregatorV3 interface

**Reply by EngrPips** [reply]
What version of the chainlink repo did you installed ?

**Reply by Joel-eth** [reply]
v0.8

**Reply by EngrPips** [reply]
run the below command in your terminal and you should successfully install the chainlink repo that have the `Chainlink's AggregatorV3 interface ` into your codebase 

```javascript
forge install smartcontractkit/chainlink-brownie-contracts --no-commit
```

That is the exact command Patrick use to install the chainlink repo that contains `Chainlink's AggregatorV3 interface ` .

I will be looking forward to hear from you

**Reply by Joel-eth** [reply]
Sorry but do you mean my system terminal or in remix?

**Reply by EngrPips** [reply]
Are you working in remix or locally on your machine?

**Reply by Joel-eth** [reply]
remix

**Comment by AbigirlMuchineripi** [comment]
Hie @Jostar-dev  I remember , I had the same challenge last week , unfortunately i cant get the link now where i got the solidity code on the chainlink website but the good news is that i saved it . Here it is : 

`// SPDX-License-Identifier: MIT
pragma solidity ^0.8.7;

import {AggregatorV3Interface} from "@chainlink/contracts@1.0.0/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";

/**
 * THIS IS AN EXAMPLE CONTRACT THAT USES HARDCODED
 * VALUES FOR CLARITY.
 * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
 * DO NOT USE THIS CODE IN PRODUCTION.
 */

/**
 * If you are reading data feeds on L2 networks, you must
 * check the latest answer from the L2 Sequencer Uptime
 * Feed to ensure that the data is accurate in the event
 * of an L2 sequencer outage. See the
 * https://docs.chain.link/data-feeds/l2-sequencer-feeds
 * page for details.
 */

contract DataConsumerV3 {
    AggregatorV3Interface internal dataFeed;

    /**
     * Network: Sepolia
     * Aggregator: BTC/USD
     * Address: 0x1b44F3514812d835EB1BDB0acB33d3fA3351Ee43
     */
    constructor() {
        dataFeed = AggregatorV3Interface(
            0x1b44F3514812d835EB1BDB0acB33d3fA3351Ee43
        );
    }

    /**
     * Returns the latest answer.
     */
    function getChainlinkDataFeedLatestAnswer() public view returns (int) {
        // prettier-ignore
        (
            /* uint80 roundID */,
            int answer,
            /*uint startedAt*/,
            /*uint timeStamp*/,
            /*uint80 answeredInRound*/
        ) = dataFeed.latestRoundData();
        return answer; // us$70,715.53077326
    }
}
`

**Reply by AbigirlMuchineripi** [reply]
I am sorry i dont know how to post my code properly but copy from SPDX...

Found the link : https://docs.chain.link/data-feeds/using-data-feeds#solidity

**Reply by EngrPips** [reply]
```javascript
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.7;

import {AggregatorV3Interface} from "@chainlink/contracts@1.0.0/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";

/**

THIS IS AN EXAMPLE CONTRACT THAT USES HARDCODED
VALUES FOR CLARITY.
THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
DO NOT USE THIS CODE IN PRODUCTION.
*/
/**

If you are reading data feeds on L2 networks, you must
check the latest answer from the L2 Sequencer Uptime
Feed to ensure that the data is accurate in the event
of an L2 sequencer outage. See the
https://docs.chain.link/data-feeds/l2-sequencer-feeds
page for details.
*/
contract DataConsumerV3 {
AggregatorV3Interface internal dataFeed;

/**
 * Network: Sepolia
 * Aggregator: BTC/USD
 * Address: 0x1b44F3514812d835EB1BDB0acB33d3fA3351Ee43
 */
constructor() {
    dataFeed = AggregatorV3Interface(
        0x1b44F3514812d835EB1BDB0acB33d3fA3351Ee43
    );
}

/**
 * Returns the latest answer.
 */
function getChainlinkDataFeedLatestAnswer() public view returns (int) {
    // prettier-ignore
    (
        /* uint80 roundID */,
        int answer,
        /*uint startedAt*/,
        /*uint timeStamp*/,
        /*uint80 answeredInRound*/
    ) = dataFeed.latestRoundData();
    return answer; // us$70,715.53077326
}
}
```

**Reply by Joel-eth** [reply]
Thanks I was able to find the source code from another repository.

**Reply by Joel-eth** [reply]
> // SPDX-License-Identifier: MIT
pragma solidity 0.8.0;
// This is the chainlink interface that we use to get the ETH price
interface IAggregatorV3 {
  function decimals() external view returns (uint8);
  function description() external view returns (string memory);
  function version() external view returns (uint256);
  function getRoundData(uint80 _roundId)
    external
    view
    returns (
      uint80 roundId,
      int256 answer,
      uint256 startedAt,
      uint256 updatedAt,
      uint80 answeredInRound
    );
  function latestRoundData()
    external
    view
    returns (
      uint80 roundId,
      int256 answer,
      uint256 startedAt,
      uint256 updatedAt,
      uint80 answeredInRound
    );
}

**Reply by EngrPips** [reply]
So your issue is fixed now, right ?

**Reply by Joel-eth** [reply]
Yeah. Thanks Bro you're the best

**Reply by EngrPips** [reply]
Anytime friend, i am glad we figured it out.
